### ISSUE TITLE
x/build/cmd/watchflakes: adjust behavior for non-default branches

### ISSUE THREAD

**Issue by bcmills** [golang/go#59483]
When `watchflakes` posts to an issue (as in https://github.com/golang/go/issues/58714#issuecomment-1448384846), it gives the timestamp and Go commit hash but not the Go branch name.

For failures on release branches that occur after an attempted fix, the branch name is a very important detail: it helps us figure out whether the fix was successful (and may need to be backported) or unsuccessful (and needs to be further revised).

Specifically, I suggest that when `watchflakes` posts newly-identified test flakes:
- [] the summary lines for failures on non-default branches should mention the branch name, and
- [] `watchflakes` should not reopen a closed issue if the only new failures are on non-default branches.

(attn @cherrymui; CC @golang/release)

**Comment by cherrymui** [comment]
Thanks. Will do.

**Comment by cherrymui** [comment]
The tricky part is that the dashboard on the subrepo doesn't have any information about the branch of the main repo. So watchflakes doesn't see it easily. (In fact, watchflakes already has the logic to print a different text, but `GoBranch` is never set...)

watchflakes could deduce a go commit's branch from the dashboard of the main repo on that branch. That will need more work.

**Comment by bcmills** [comment]
Yep. The other options would be to change the dashboard to populate more data, or to have `watchflakes` separately load data from `maintner` (as I did for `greplogs --triage` back when).
